build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts

```diff
@@ -226,9 +227,37 @@ export class GenVerTask extends LeafTask {
 	}
 }
 
-export class TypeValidationTask extends LeafWithDoneFileTask {
-	protected async getDoneFileContent(): Promise<string | undefined> {
-		return JSON.stringify(this.package.packageJson);
+export class TypeValidationTask extends LeafWithFileStatDoneFileTask {
+	private inputFiles: string[] | undefined;
+	private outputFiles: string[] | undefined;
+
+	/**
+	 * All config for the type tests is contained in package.json.
+	 */
+	protected async getInputFiles(): Promise<string[]> {
+		if (this.inputFiles === undefined) {
+			const { packageJsonPath: previousPackageJsonPath } = getTypeTestPreviousPackageDetails(
+				this.node.pkg,
+			);
```

Comment:
This breaks build incrementality when typetests are disabled as the file (probably) can't be found and stats aren't retrieved.
Additionally, this really is a `TscDependentTask` as any of that changing can change the API.
Fix proposed in #20986 
cc: @alexvy86